src/corehost/cli/fxr_resolver.h

```diff
@@ -77,21 +77,27 @@ int load_fxr_and_get_delegate(hostfxr_delegate_type type, THostPathToConfigCallb
         dotnet_root.c_str()
     };
 
-    hostfxr_handle context;
-    int rc = hostfxr_initialize_for_runtime_config(config_path.c_str(), &parameters, &context);
-    if (!STATUS_CODE_SUCCEEDED(rc))
-        return rc;
-
-    rc = hostfxr_get_runtime_delegate(context, type, reinterpret_cast<void**>(delegate));
+    hostfxr_set_error_writer_fn set_error_writer_fn = reinterpret_cast<hostfxr_set_error_writer_fn>(pal::get_symbol(fxr, "hostfxr_set_error_writer"));
 
-    int rcClose = hostfxr_close(context);
-    if (rcClose != StatusCode::Success)
     {
-        assert(false && "Failed to close host context");
-        trace::verbose(_X("Failed to close host context: 0x%x"), rcClose);
-    }
+        propagate_error_writer_t propagate_error_writer_to_hostfxr(set_error_writer_fn);
```

Comment:
Should we consider this for 3.0 - or at least 3.1?

Reply:
I think this definitely makes the 3.1 bar and probably could be done in 3.0.

@vitek-karas thoughts?

Reply:
We should give it a try for 3.0 till we can. @jeffschwMSFT are you OK with porting this fix to 3.0 branch?

The bug is that in COM scenarios errors from hosting layer about missing frameworks and such (basically anything from hostfxr) would only be reported as error codes, no text. With the fix we will be able to propagate the full error information at least in COM case. The same code path is used by IJW and by WinRT, but those don't have a good mechanism to report the error description.